### PR TITLE
Floats in exponent notation

### PR DESCRIPTION
**Pull request by R4PaSs**
As a new addition to the parser, as requested in #1261, literal floating-point values in exponent notation.

### CODE REVIEW

**Comment by Morriar** [comment]
+1

**Comment by R4PaSs** [comment]
@privat I could, but would it simplify the whole process ? I mean we already do such a thing for int literals with the different bases, it simplifies the code in literal, why do it differently in Float then ?

**Comment by privat** [comment]
The separation is more acceptable with integer because they have a specific notation with refused/accepted characters.
With floats, one can define a common more general token with an optional exponent part, therefore having two tokens make less sense.

**Comment by xymus** [comment]
+1

**Comment by R4PaSs** [comment]
Reroll: Float with expoenents are now a facultative exponent part, good call @privat, this removes a lot of code !

File: src/parser/nit.sablecc3xx [hunk 1]
```diff
@@ -207,7 +207,7 @@ bytenum = digit+ 'u8';
 hex_bytenum = ('0x' | '0X') hexdigit+ 'u8';
 bin_bytenum = ('0b' | '0B') bindigit+ 'u8';
 oct_bytenum = ('0o' | '0O') octdigit+ 'u8';
-float = digit* '.' digit+;
+float = digit* '.' digit+ ( | ('E'|'e') ('+'|'-'|) digit+);
```
Comment (privat) [inline src/parser/nit.sablecc3xx]: not yet since `1e10` will be refused as a valid Float and that is not pola

**Comment by R4PaSs** [comment]
@privat Admittedly, I forgot that, this should be fixed with this reroll

**Comment by privat** [comment]
+1

**Pull request merged**